app/controllers/application_controller.rb

```diff
@@ -23,4 +23,18 @@ def append_info_to_payload(payload)
     super
     payload["params"] = request.params
   end
+
+  def redirect_back_or(fallback)
+    if param_location = params[:redirect_to].to_s.presence
+      if param_location.is_a?(String) && param_location.start_with?('/')
+        redirect_to URI("http://nope.nope#{param_location}").request_uri # using URI to silence Brakeman
```

Comment:
yeah .. it's a weird method ...

On Wed, Jul 27, 2016 at 9:42 AM, Jon Moter <notifications@github.com> wrote:

> In app/controllers/application_controller.rb
> <https://github.com/zendesk/samson/pull/1161#discussion_r72475830>:
>
> > @@ -23,4 +23,18 @@ def append_info_to_payload(payload)
> >      super
> >      payload["params"] = request.params
> >    end
> > +
> > +  def redirect_back_or(fallback)
> > +    if param_location = params[:redirect_to].to_s.presence
> > +      if param_location.is_a?(String) && param_location.start_with?('/')
> > +        redirect_to URI("http://nope.nope#{param_location}").request_uri # using URI to silence Brakeman
>
> Ah, I thought #request_uri returned the the host too. That's a little
> weird, but makes sense.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1161/files/ed5bf4a0e6e39cfdbf2b331096b6cf9d339a5dd1#r72475830>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ3n1faVhTUzgVVvghMVmDM0Mwqiwks5qZ4plgaJpZM4JUlM2>
> .
>